rflx/cli.py

```diff
@@ -314,25 +321,25 @@ def graph(args: argparse.Namespace) -> None:
         fail(f'directory not found: "{args.output_directory}"', Subsystem.CLI)
 
     model, _ = parse(args.files, args.no_verification)
-    locations: Dict[str, Dict[str, Dict[str, Dict[str, int]]]] = defaultdict(dict)
-
-    for m in [*model.messages, *model.sessions]:
-        assert isinstance(m, (Message, Session))
-        name = m.identifier.flat
-        filename = args.output_directory.joinpath(name).with_suffix(f".{args.format}")
-        Graph(m).write(filename, fmt=args.format)
 
-        assert m.location
-        assert m.location.start
-        assert m.location.end
-        assert m.location.source
+    for m in model.messages:
+        filename = args.output_directory.joinpath(m.identifier.flat).with_suffix(f".{args.format}")
+        write_graph(create_message_graph(m), filename, fmt=args.format)
 
-        source = str(m.location.source)
+    for s in model.sessions:
+        filename = args.output_directory.joinpath(s.identifier.flat).with_suffix(f".{args.format}")
+        write_graph(create_session_graph(s, args.ignore), filename, fmt=args.format)
```

Comment:
This could be a good use case for a single dispatch function to remove the code duplication. But I'm also fine with keeping the code as it is.

Reply:
Not this time. I'm not certain if that would actually improve the overall comprehensibility of the code.